contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,850 @@
+
+    /**
+     * @dev Claim the undelegated BNB from the pool after unbondPeriod
+     */
+    function claim(
+        address operatorAddress,
+        uint256 requestNumber
+    ) external whenNotPaused validatorExist(operatorAddress) {
+        uint256 bnbAmount = IStakeCredit(_validators[operatorAddress].creditContract).claim(msg.sender, requestNumber);
+        emit Claimed(operatorAddress, msg.sender, bnbAmount);
+    }
+
+    function sync(address[] calldata operatorAddresses, address account) external whenNotPaused {
+        uint256 _length = operatorAddresses.length;
+        address[] memory validatorPools = new address[](_length);
+        address _pool;
+        for (uint256 i = 0; i < _length; ++i) {
+            _pool = _validators[operatorAddresses[i]].creditContract;
+            require(_pool != address(0), "validator not exist");
+            validatorPools[i] = _pool;
+        }
+
+        IGovBNB(GOV_TOKEN_ADDR).sync(validatorPools, account);
+    }
+
+    /*----------------- system functions -----------------*/
+    function distributeReward(address consensusAddress) external payable onlyValidatorContract {
+        address operatorAddress = _consensusToOperator[consensusAddress];
+        Validator memory valInfo = _validators[operatorAddress];
+        if (valInfo.creditContract == address(0) || valInfo.jailed) {
+            emit RewardDistributeFailed(operatorAddress, "INVALID_VALIDATOR");
+            return;
+        }
+
+        try IStakeCredit(valInfo.creditContract).distributeReward{ value: msg.value }(valInfo.commission.rate) {
+            emit RewardDistributed(operatorAddress, msg.value);
+        } catch Error(string memory reason) {
+            emit RewardDistributeFailed(operatorAddress, bytes(reason));
+        } catch (bytes memory lowLevelData) {
+            emit RewardDistributeFailed(operatorAddress, lowLevelData);
+        }
+    }
+
+    /**
+     * @dev Get new eligible validators from consensus engine
+     */
+    function updateEligibleValidatorSet(
+        address[] calldata validators,
+        uint64[] calldata votingPowers,
+        bytes[] calldata voteAddrs
+    ) external onlyCoinbase onlyZeroGasPrice {
+        uint256 newLength = validators.length;
+        if (newLength == 0) {
+            return;
+        }
+        uint256 oldLength = _eligibleValidatorSet.length;
+        if (oldLength > newLength) {
+            for (uint256 i = newLength; i < oldLength; ++i) {
+                _eligibleValidatorSet.pop();
+            }
+        }
+
+        uint256 j;
+        IBSCValidatorSet.Validator[] memory bscValidatorSet = new IBSCValidatorSet.Validator[](newLength);
+        for (uint256 i; i < newLength; ++i) {
+            if (j >= oldLength) {
+                _eligibleValidatorSet.push(validators[i]);
+            } else {
+                _eligibleValidatorSet[j] = validators[i];
+            }
+            ++j;
+
+            bscValidatorSet[i] = IBSCValidatorSet.Validator({
```

Comment:
I am thinking that: the `stakeHub` don need to have this `_eligibleValidatorSet`.  

By calling `updateValidatorSetV2`, the validator set contract can decide to tempary store the `bscValidatorSet` here.

Reply:
Done